<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1263</width>
    <height>899</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#centralwidget {
	/*background: #dee0e0;*/
}

QWidget {
	/*background: rgb(37, 38, 49);*/	
	color:white;
	font-size: 8pt
}
QGroupBox {
	/*background: #15242d;*/
	/*color:rgb(241, 241, 241);*/
	color: #49505c;
}

QComboBox{
	background:#49505c;
	color: white;
}

QPushButton {
	background:#49505c;
	color:rgb(241, 241, 241);
	border-radius: 8px;
	height:30px
}
QLabel{
color: #49505c;
}

QCheckBox{
color: #49505c;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QGroupBox" name="grpbx_DigitalFilter">
      <property name="title">
       <string>Digital Filter Design</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="2">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="PlotWidget" name="magnitudeGraph" native="true">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="PlotWidget" name="phaseGraph" native="true">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <widget class="QWidget" name="wgt_unitCircle" native="true">
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="1" column="2" alignment="Qt::AlignLeft">
           <widget class="QCheckBox" name="pair_mode_toggle">
            <property name="maximumSize">
             <size>
              <width>70</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Pair</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" rowspan="3">
           <widget class="QWidget" name="wgt_buttons" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="leftMargin">
              <number>10</number>
             </property>
             <item>
              <widget class="QGroupBox" name="groupBox_4">
               <property name="minimumSize">
                <size>
                 <width>110</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>130</height>
                </size>
               </property>
               <property name="title">
                <string>Mode</string>
               </property>
               <widget class="QRadioButton" name="add_radioButton">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>30</y>
                  <width>111</width>
                  <height>23</height>
                 </rect>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:black;</string>
                </property>
                <property name="text">
                 <string>Add Mode</string>
                </property>
               </widget>
               <widget class="QRadioButton" name="move_radioButton">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>59</y>
                  <width>111</width>
                  <height>23</height>
                 </rect>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:black;</string>
                </property>
                <property name="text">
                 <string>Move Mode</string>
                </property>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_2">
               <property name="minimumSize">
                <size>
                 <width>110</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>130</height>
                </size>
               </property>
               <property name="title">
                <string>Poles</string>
               </property>
               <widget class="QPushButton" name="btn_addPoles">
                <property name="geometry">
                 <rect>
                  <x>5</x>
                  <y>24</y>
                  <width>100</width>
                  <height>30</height>
                 </rect>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>100</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">
font: 7.8pt ;</string>
                </property>
                <property name="text">
                 <string>Add Poles</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">btngrp_zerosPoles</string>
                </attribute>
               </widget>
               <widget class="QPushButton" name="btn_removePoles">
                <property name="geometry">
                 <rect>
                  <x>5</x>
                  <y>62</y>
                  <width>100</width>
                  <height>31</height>
                 </rect>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>100</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">font: 7.7pt ;</string>
                </property>
                <property name="text">
                 <string>Remove Poles</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">btngrp_zerosPoles</string>
                </attribute>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_3">
               <property name="minimumSize">
                <size>
                 <width>110</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>130</height>
                </size>
               </property>
               <property name="title">
                <string>Zeros</string>
               </property>
               <widget class="QPushButton" name="btn_addZeros">
                <property name="geometry">
                 <rect>
                  <x>5</x>
                  <y>23</y>
                  <width>100</width>
                  <height>30</height>
                 </rect>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>100</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">font: 7.8pt ;</string>
                </property>
                <property name="text">
                 <string>Add Zeros</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">btngrp_zerosPoles</string>
                </attribute>
               </widget>
               <widget class="QPushButton" name="btn_RemoveZeros">
                <property name="geometry">
                 <rect>
                  <x>5</x>
                  <y>61</y>
                  <width>100</width>
                  <height>30</height>
                 </rect>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>100</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">font: 7.7pt ;</string>
                </property>
                <property name="text">
                 <string>Remove Zeros</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">btngrp_zerosPoles</string>
                </attribute>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_removeAll">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Remove All</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="PlotWidget" name="unitCircleGraph" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>400</width>
           <height>400</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>400</width>
           <height>400</height>
          </size>
         </property>
         <property name="sizeIncrement">
          <size>
           <width>1</width>
           <height>1</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
        </widget>
       </item>
       <item row="0" column="0" alignment="Qt::AlignHCenter">
        <widget class="QPushButton" name="importButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">width: 95px</string>
         </property>
         <property name="text">
          <string>Import Filter</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" alignment="Qt::AlignHCenter">
        <widget class="QPushButton" name="exportButton">
         <property name="styleSheet">
          <string notr="true">width: 95px;</string>
         </property>
         <property name="text">
          <string>Export Filter</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="grpbx_AllPassFilter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>400</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Custom Phase Correction</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="3" column="0">
        <widget class="PlotWidget" name="allPassGraph" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>70</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>300</width>
           <height>16777215</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>160</height>
          </size>
         </property>
         <property name="title">
          <string>All Pass Filters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTableWidget" name="table_coeff">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>150</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">color:black</string>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::DoubleClicked</set>
            </property>
            <property name="alternatingRowColors">
             <bool>false</bool>
            </property>
            <property name="showGrid">
             <bool>true</bool>
            </property>
            <property name="cornerButtonEnabled">
             <bool>false</bool>
            </property>
            <property name="columnCount">
             <number>1</number>
            </property>
            <attribute name="horizontalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <column/>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QWidget" name="wgt_coefficient" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>300</width>
           <height>70</height>
          </size>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>value of a</string>
            </property>
            <property name="buddy">
             <cstring>wgt_coefficient</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QComboBox" name="comboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>27</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">	background:#49505c;
	color: white;</string>
            </property>
            <property name="editable">
             <bool>true</bool>
            </property>
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>-1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>-0.9</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>-0.495</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0.99</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0.345</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="btn_addCoeff">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="btn_removeCoeff">
            <property name="styleSheet">
             <string notr="true">width:65px</string>
            </property>
            <property name="text">
             <string>Remove</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0" alignment="Qt::AlignHCenter">
        <widget class="QCheckBox" name="all_pass_enable">
         <property name="styleSheet">
          <string notr="true">font-size: 8pt</string>
         </property>
         <property name="text">
          <string>All pass filter</string>
         </property>
         <property name="iconSize">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="grpbx_RealtimeFiltering">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>300</height>
       </size>
      </property>
      <property name="title">
       <string>Realtime Filtering</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="2" column="0" rowspan="2">
        <widget class="QPushButton" name="btn_play">
         <property name="text">
          <string>Play</string>
         </property>
         <property name="shortcut">
          <string>Space</string>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="lbl_speed">
         <property name="maximumSize">
          <size>
           <width>150</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Speed: 1</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>speed_slider</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QSlider" name="speed_slider">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">
QSlider::handle:horizontal {
    background-color:rgb(50, 131, 179);
    height: 40px;
    width: 40px;

border-radius: 5px
    }</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2">
        <widget class="PlotWidget" name="filteredGraph" native="true"/>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="PlotWidget" name="InputGraph" native="true"/>
       </item>
       <item row="1" column="5">
        <widget class="PlotWidget" name="mouseInputGraph" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>166</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QCheckBox" name="mouse_en">
         <property name="text">
          <string>Mouse Enable</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="btn_openFile">
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Choose File</string>
         </property>
         <property name="shortcut">
          <string>Ctrl+O</string>
         </property>
        </widget>
       </item>
       <item row="3" column="5">
        <widget class="QPushButton" name="btnClr">
         <property name="maximumSize">
          <size>
           <width>250</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">width:150px</string>
         </property>
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1263</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionImport">
   <property name="text">
    <string>Import Zeros/Poles</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>Export Zeros/Poles</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="btngrp_zerosPoles"/>
 </buttongroups>
</ui>
